pkg/operator/loglevel/controller.go

```diff
@@ -0,0 +1,113 @@
+
+	log "github.com/sirupsen/logrus"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	logLevelUtils "github.com/openshift/library-go/pkg/operator/loglevel"
+
+	"github.com/openshift/cloud-credential-operator/pkg/operator/status"
+	"github.com/openshift/cloud-credential-operator/pkg/operator/utils"
+)
+
+const (
+	controllerName = "loglevel"
+)
+
+// Add creates a new ConfigController and adds it to the Manager.
+func Add(mgr manager.Manager, kubeConfig string) error {
+	r := newReconciler(mgr)
+
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to CloudCredential object and reconcile the loglevel changes
+	err = c.Watch(
+		&source.Kind{Type: &operatorv1.CloudCredential{}},
+		&handler.EnqueueRequestForObject{})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	r := &ReconcileCloudCredentialConfig{
+		Client: mgr.GetClient(),
+	}
+	status.AddHandler(controllerName, r)
+
+	return r
+}
+
+var _ reconcile.Reconciler = &ReconcileCloudCredentialConfig{}
+
+// ReconcileCloudCredentialConfig reconciles a CredentialConfig object
+type ReconcileCloudCredentialConfig struct {
+	client.Client
+}
+
+// Reconcile reads the state of the cluster for a CredentialConfig object and
+// makes changes based on the state read and what is in the CredentialConfig.LogLevel
+// Automatically generate RBAC rules to allow the Controller to read and write required types.
+// +kubebuilder:rbac:groups=operator.openshift.io,resources=cloudcredential/spec,verbs=get
+func (r *ReconcileCloudCredentialConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	logger := log.WithFields(log.Fields{
+		"controller": controllerName,
+		"cr":         fmt.Sprintf("%s/%s", request.NamespacedName.Namespace, request.NamespacedName.Name),
+	})
+
+	// GetOperatorLogLevel
+	log.Debug("Checking log level for Reconcile request")
+	desiredLogLevel, err := utils.GetOperatorLogLevel(r, logger)
+
+	// Error with the current logLevel
+	if err != nil || !logLevelUtils.ValidLogLevel(desiredLogLevel) {
+		return reconcile.Result{Requeue: false}, fmt.Errorf("the loglevel defined for the cloudcredential object is invalid")
+	}
+
+	currentLogLevel, errBool := logLevelUtils.GetLogLevel()
+	if errBool {
+		log.Debug("Unable to get the current loglevel. Defaulting to 'Normal'")
+	}
+
+	if currentLogLevel == desiredLogLevel {
+		return reconcile.Result{Requeue: false}, nil
+	}
+
+	// Set logLevel to the new level
+	logLevelUtils.SetLogLEvel(desiredLogLevel)
```

Comment:
```suggestion
	logLevelUtils.SetLogLevel(desiredLogLevel)
```

Reply:
This function returns an error which needs to be handled

Reply:
I've updated this to catch the error, but I have not changed the function name. There's a typo in the library-go version used.
https://github.com/openshift/cloud-credential-operator/pull/379/files#diff-e5b905c5074fe810ed7617bc85a7128c651866094cb4934536459b932ce09236R65

I didn't want my first PR to contain updates to the library versions, but if you would prefer I can update the version of library-go.

The PR this was changed is here: 
https://github.com/openshift/library-go/commit/15d11d2f6bfcbe15679acd184ac69c77aa2e65bc

Reply:
I am not sure if this deserves a library-go version update. I am leaning towards no. @joelddiaz wdyt?